lib/SieveKeywordRegistry.php

```diff
@@ -80,8 +96,44 @@ public function activate($extension)
             return;
         }
 
+        if (isset($this->requiredExtensions[$extension])) {
+            // already loaded
+            return;
+        } else {
+            $this->requiredExtensions[$extension] = true;
+        }
+
+        if (isset($this->forbiddenExtensions[$extension])) {
+            trigger_error(
+                "Extension $extension cannot be loaded together with the extensions " . implode(
+                    ', ',
+                    $this->forbiddenExtensions[$extension]
+                )
+            );
+        }
+
         $xml = $this->registry_[$extension];
 
+        if (isset($xml["require"])) {
+            $requireExtensions = explode(',', $xml["require"]);
+            foreach ($requireExtensions as $require) {
+                if ($require[0] !== '!') {
+                    if (!isset($this->requiredExtensions[$require])) {
+                        trigger_error("Extension $extension requires the extension $require to be loaded.");
```

Comment:
This means that `require ["extensionRequiringA", "ExtensionA"]` will fail but `require ["ExtensionA", "extensionRequiringA"]`, succeed right?